### DISCUSSION TITLE
Query resolver gets called multiple times (2) per single request.

### DISCUSSION THREAD

**Discussion by ErnestoBorio** [MichalLytek/type-graphql#992] [Q&A] [answered]
I'm using type-graphql and Apollo, the client makes 1 POST request with the graphql query and my backend runs the resolver 2 times. This is not my main concern, although I'd like to know why, but the thing is that **only the last of the 2 times it gets the cookies**, which are essential to determine if the client is authorized to view the content or not. 
The only response the client picks up is from one of the requests that didn't pass the cookie through so it gets the error response. 

```typescript
  @Resolver(ContentModel)
  export class ContentSchemaResolver {
    @Query(() => ContentModel, { nullable: true })
    async content(@Ctx() context: HttpContext, @Arg('id', () => ID) id: string) {
      const content = await ContentModel.findOne(id);
      // [...]
      return content;
    }
    // [...]
  }
```
_PS: Should I paste more code?_

**Comment by MichalLytek** [comment, marked as answer]
That doesn't look like a TypeGraphQL issue. It builds a GraphQL schema that is then called by Apollo Server (or other lib).

**Reply by ErnestoBorio** [reply]
Ah ok, thanks.
